### PR TITLE
Andrew onboarding refactor

### PR DESCRIPTION
**Pull request by ahamacher**
This refactor removes Maxwell from the onboarding process as well as changes the way the onboarding components work.

All onboarding questions now require a react component to handle the question. This allows us to have any number of fields on an onboarding tile.

Fix the dashboard display as the information from our backend was being received at a different time than required with the refactor changes.

### CODE REVIEW

**Review by JibbyAni — CHANGES_REQUESTED** [review]
Left a few comments! Once changes are made tag me for a re-review and I'll take another look over

File: src/components/header/header.js [hunk 1]
```diff
@@ -35,6 +34,7 @@ const Header = props => {
   const logoutUser = () => {
     const { logout } = props;
     logout();
+    window.location.href = "http://localhost:3000/"
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/header/header.js:37]: This is being addressed in a future PR. It no longer uses an exact route and instead uses the relative route through the react-router.

File: src/utils/validate.js [hunk 2]
```diff
@@ -3,113 +3,103 @@ import * as onboardingUtil from './onboardingUtil';
...
 }
 
-const checkEmail = (email) => {
-  return new Promise((res, rej) => {
-    if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email))
-  {
-    res(true)
-  } else {
-    rej(false)
+export const passwordValidation = (passwords) => {
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/utils/validate.js:40]: Totally agree on a more secure password, I actually think we should force more than just 1 number and 1 character, could we make it so it is 1 number, 1 symbol and at least 1 uppercase and lowercase letter? 

Could we leave this as it is now as it makes it easier for testing to have a simple password when I am making and entering in 10-20 accounts per day. I was thinking an updated verification could be included in the forgot password Trello tile.

File: src/components/confirmation/confirmation.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    })
+};
+
+const mapDispatchToProps = dispatch => ({
+    closeModal: () => dispatch(closeModal()),
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(ConfirmationModal);
+
+// data flow diagram for bloom
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/confirmation/confirmation.js:144]: these were missed in deletion of my personal notes on how i wanted to construct the component. deleted.

File: src/utils/profileApiUtil.js [hunk 4]
```diff
@@ -11,3 +11,7 @@ export const getProfile = () => {
 export const newProfile = payload => {
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/utils/profileApiUtil.js:11]: This api route is never used as I use the one in userApiUtils to create a new customer. Deleted this and all references / imports as they were never called in the components.

File: src/utils/profileApiUtil.js [hunk 5]
```diff
@@ -11,3 +11,7 @@ export const getProfile = () => {
...
   return axios.post('/api/user', payload, { withCredentials: true })
 };
+
+export const getAddress = () => {
+  return axios.get(`api/profile/address`, { withCredentials: true })
+};
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/utils/profileApiUtil.js:17]: Get profile gets all of a profiles attributes, we dont need to specify the ID as the id is encrypted into the session cookie provided by firebase and is decoded on our backend. 

updated all sendProfiles with newProfile

File: src/components/dashboard/dashTiles/addressModule.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    return (
+        <StyledAddress>
+            {address}
+        </StyledAddress>
+    )
+}
+
+const mSTP = state => {
+    //TODO: delete this:
+    let address = "Test Address Delete Before Push";
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/dashboard/dashTiles/addressModule.js:22]: deleted.

File: src/components/dashboard/dashTiles/valuationModule.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                    {confidence}
+                </StyledConfidence>
+            </StyledValueInfo>
+        </>
+    )
+}
+
+const mSTP = state => {
+    //TODO: delete this:
+    let valuation = "Test valuation Delete Before Push";
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/dashboard/dashTiles/valuationModule.js:73]: deleted.

File: src/components/dashboard/dashboard.js [hunk 8]
```diff
@@ -1,11 +1,51 @@
...
+import { connect } from 'react-redux';
+import { StyledContainerCenter, StyledTile } from './dashboardStyle';
+import AddressModule from './dashTiles/addressModule';
+import ValuationModule from './dashTiles/valuationModule';
+import { loadModal } from '../../actions/modalActions'
+import { getAsset } from '../../actions/assetProfileActions';
 
 const Dashboard = props => {
+  useEffect(() => {
+    console.log("hit the dashboard use effect");
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/dashboard/dashboard.js:11]: This was just testing to see if my component was loading correctly, and if the useEffect was being properly called on a component load. It has no further purposed and has been deleted.

File: src/utils/verifyUtil.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios';
+
+// TODO: implement resending of codes
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/utils/verifyUtil.js:3]: deleted.

File: src/utils/validate.js [hunk 10]
```diff
@@ -3,113 +3,103 @@ import * as onboardingUtil from './onboardingUtil';
...
-  error: errors.onboarding.phoneNumberError,
-  validate: phoneNumber,
+// reworked validators:
+// validator should take an input and then return in the same format (string, or object) an error or null
+// null being returned would mean there is no error
+// a string or object being returned would mean there is an error and then the response could be passed to the necessary fields
+
+export const ssnValidation = (social) => {
+  const { onboarding: { SSN } } = errors;
+  if (social.length !== 4) {
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/utils/validate.js:13]: at the time of this PR we were still only seeking the last 4 of a SSN, the change for a full SSN is in a future PR.

File: src/utils/assetProfileUtil.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios';
+
+export const newAssetProfile = () => {
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/utils/assetProfileUtil.js:3]: Added in comments but here as well. An asset profile is all the information that pertains to a specific Asset or in other terms a customer's property.

File: src/components/splash/test.js [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+      setText(inputText);
+  }
+
+  const resendOTP = () => {
+    axios.get('/api/pricing/maxotpresend').then(
+      res => console.log(res)
+    )
+  }
+
+  const getPersonInfo = () => {
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/splash/test.js:27]: deleted, but left the component. I use this component when i am building out new features to test the routes from the "splash" page. It is something that will be removed once we have an actual splash page to show, for now I have deleted the references to maxwell though.

File: src/components/session/verify.js [hunk 13]
```diff
@@ -1,55 +1,78 @@
-import React, { useEffect } from 'react';
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/session/verify.js:1]: This whole component is replaced in a future PR (repurposed for bloom KBA). Replacing it here will make more work in fixing all the conflicts in the future.

File: src/constants/constants.js [hunk 14]
```diff
@@ -17,9 +17,22 @@ const constants = {
         goals: "What are some anticipated life goals and changes over the next 5 years",
         phoneNumber: "What's your phone number?",
         zipcode: "What is your current zipcode?",
+        address: "Enter your address",
         ssn: "What are the last 4 digits of your SSN?",
         duration: "How long do you plan on staying in your current home?",
         amountNeeded: "How much do you need for your upcoming goal(s)?",
+        maxwellOTP: "Please enter the One Time Passcode that was sent to the phone-number provided",
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/constants/constants.js:24]: deleted.

File: src/components/dashboard/dashTiles/valuationModule.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+import {  } from '../../../actions/profileActions';
+
+const valuationText = {
+    lastUpdated: "Estimated Home Value as of ",
+    confidenceLevel: "Confidence Level ",
+}
+
+const ValuationModule = props => {
+    const { valuation, confidence, lastUpdated } = props;
+    // useEffect(() => {
```
[the remark this answers is not in the document]
Comment (ahamacher) [reply src/components/dashboard/dashTiles/valuationModule.js:14]: deleted.

**Review by JibbyAni — APPROVED** [review]
Looks good!

**Pull request merged**